### PR TITLE
Properly parse code in Performance/Count that extends enumerable

### LINKED ISSUE #2080: Bug in Performance/Count
With this file `test.rb`:
```rb
# encoding: utf-8

# nodoc #
class A < Array
  def count(&block)
    select(&block).count
  end
end

a = A.new
a += [1, 2, 3, 4, 5, 6]
p a.count { |e| e > 1 }
```

RuboCop tells me that `An error occurred while Performance/Count cop was inspecting /Users/bquorning/Code/rubocop/test.rb.`

Extra information:

    0.32.1 (using Parser 2.2.2.5, running on ruby 2.2.2 x86_64-darwin13)

Actually, I was using the latest master, cdfb667243136b173ddbac804ded35a48d5ec53a.

### PR DESCRIPTION
**Pull request by rrosenblum**
This fixes #2080

### CODE REVIEW

File: spec/rubocop/cop/performance/count_spec.rb [hunk 1]
```diff
@@ -86,6 +86,20 @@
       expect(cop.highlights).to eq(["#{selector}(&:something).count"])
     end
 
+    it "registers an offense for #{selector}(&:something).count " \
```
Comment (bbatsov) [inline spec/rubocop/cop/performance/count_spec.rb]: I don't see anything in the changed code that take into account the superclass, so this description isn't precise IMO.

File: spec/rubocop/cop/performance/count_spec.rb [hunk 2]
```diff
@@ -86,6 +86,20 @@
...
+       'called on a class that includes or extends an enumerable' do
+      source = ['class A < Array',
+                '  def count(&block)',
+                "    #{selector}(&block).count",
+                '  end',
+                'end']
+      inspect_source(cop, source)
+
+      expect(cop.messages)
+        .to eq(["Use `count` instead of `#{selector}...count`."])
```
Comment (bbatsov) [inline spec/rubocop/cop/performance/count_spec.rb]: This advise might trip some people into stackoverflow problems, but I guess that's preferable to an RuboCop error.

Comment (rrosenblum) [inline spec/rubocop/cop/performance/count_spec.rb]: The main confusion here would be caused by someone defining `count` to use `select..count` which itself seems weird and should be unnecessary. The auto-correct here would result in a recursive call. We could probably check for a method that calls itself with the same params, but this seems like an edge case. I mentioned this to @bquorning in #2080. If the outer method was named anything else, this wouldn't be an issue.

File: spec/rubocop/cop/performance/count_spec.rb (hunk 1 above)
Comment (rrosenblum) [inline spec/rubocop/cop/performance/count_spec.rb]: I modified the description. I wasn't fully sure how to describe this situation.

File: spec/rubocop/cop/performance/count_spec.rb (hunk 2 above)
Comment (bbatsov) [inline spec/rubocop/cop/performance/count_spec.rb]: OK, guess that's good enough for now.

**Pull request merged**